### PR TITLE
Feature/203 reviews backend

### PR DESCRIPTION
**Pull request by todd-dembrey**
@zerolab Updates slightly the work you did. including the titles onto the template and a slight refactor of the form code.

### CODE REVIEW

File: opentech/apply/review/forms.py [hunk 1]
```diff
@@ -84,7 +84,10 @@ def validate_unique(self):
         self.instance.author = self.request.user
         self.instance.review = self.cleaned_data
 
-        super().validate_unique()
+        try:
+            self.instance.validate_unique()
+        except ValidationError as e:
+            self._update_errors(e)
```
Comment (zerolab) [inline opentech/apply/review/forms.py]: :)

I went through the same path, first 59e5d21791c9 then this

File: opentech/apply/review/forms.py [hunk 2]
```diff
@@ -78,45 +79,29 @@ def __init__(self, *args, **kwargs):
...
-            self.instance.recommendation = self.cleaned_data['recommendation']
-        except KeyError:
-            pass
+        for field in self.get_score_fields():
+            score = int(self.cleaned_data.get(field, NA))
+            if score != NA:
+                scores.append(score)
 
-        super().save()
+        return sum(scores) / len(scores)
```
Comment (zerolab) [inline opentech/apply/review/forms.py]: so what happens when there are no score fields, thus `len(scores)` is 0?

File: opentech/apply/review/forms.py [hunk 3]
```diff
@@ -78,45 +79,29 @@ def __init__(self, *args, **kwargs):
...
-            self.instance.validate_unique()
-        except ValidationError as e:
-            self._update_errors(e)
+        super().validate_unique()
 
     def save(self, commit=True):
+        self.instance.score = self.calculate_score()
 
-        score_fields = self.get_score_fields()
+        self.instance.recommendation = self.cleaned_data['recommendation']
```
Comment (zerolab) [inline opentech/apply/review/forms.py]: I like to be defensive.

File: opentech/apply/review/views.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+from opentech.apply.funds.models import ApplicationSubmission
+from .forms import ConceptReviewForm, ProposalReviewForm
+from .models import Review
+
+
+class ReviewCreateView(CreateView):
+    model = Review
+
+    def dispatch(self, request, *args, **kwargs):
```
Comment (zerolab) [inline opentech/apply/review/views.py]: 👍

File: opentech/apply/review/forms.py (hunk 3 above)
Comment (todd-dembrey) [reply opentech/apply/review/forms.py]: `recommendation` is required. if it isnt in `cleaned_data` we have bigger problems

File: opentech/apply/review/forms.py (hunk 2 above)
Comment (todd-dembrey) [reply opentech/apply/review/forms.py]: I was missed it out. good spot.

**Pull request merged**